include/nil/actor/zk/snark/arithmetization/plonk/constraint.hpp

```diff
@@ -138,27 +134,21 @@ namespace nil {
                                     assignment =
                                         math::polynomial_shift(assignment, domain->get_domain_element(var.rotation));
                                 }
-
-                                term_value = term_value * assignment;
-                            }
-                            acc = acc + term_value;
-                        }
-                        return make_ready_future<math::polynomial<typename VariableType::assignment_type>>(acc);
+                                return assignment;
+                            });
+                        return evaluator.evaluate().get();

```

Comment:
Hm, I think I don't understand how it's work... 

Function return `future<math::polynomial<typename VariableType::assignment_type>>`
but we return `return evaluator.evaluate().get();` so, if we just return evaluator.evaluate() it's be already future? So we don't need to .get().